collect_app/src/main/java/org/odk/collect/android/activities/MainMenuActivity.java

```diff
@@ -445,61 +463,71 @@ public void onClick(DialogInterface dialog, int i) {
         alertDialog.show();
     }
 
-    @Override
-    protected Dialog onCreateDialog(int id) {
-        switch (id) {
-            case PASSWORD_DIALOG:
+    private Dialog createPasswordDialog(Intent intent) {
```

Comment:
#3623 adds some nice abstractions for this admin dialog (https://github.com/opendatakit/collect/pull/3623/files#diff-587fa336330bb907832356ee77ed08d1R405). I wasn't sure when each of these two PRs would land and how the conflict would best be dealt with. As luck would have it, they're ready at the same time. Since it's going through quality assurance right now, let's plan on merging it first and then you'll unfortunately have conflicts to fix throughout this class.